packages/chrono-core/src/processors/simple-processor.ts

```diff
@@ -12,47 +19,58 @@ export class SimpleProcessor<TaskKind, TaskData, DatastoreOptions> extends Event
   private stopRequested = false;
   private maxConcurrency: number;
 
-  constructor(input: CreateProcessorInput<TaskKind, TaskData, DatastoreOptions>) {
+  readonly claimIntervalMs = 150;
+  readonly idleIntervalMs = 5_000;
+
+  constructor(config: SimpleProcessorConfig<TaskKind, TaskData, DatastoreOptions>) {
     super();
 
-    this.datastore = input.datastore;
-    this.handler = input.handler;
-    this.maxConcurrency = input.configuration.maxConcurrency;
-    this.taskKind = input.kind;
+    this.datastore = config.datastore;
+    this.handler = config.handler;
+    this.maxConcurrency = config.maxConcurrency;
+    this.taskKind = config.kind;
   }
 
-  start(): void {
-    if (this.runningTasks.length >= 0) {
+  async start(): Promise<void> {
+    if (this.stopRequested || this.runningTasks.length >= 0) {
       return;
     }
 
-    this.stopRequested = false;
-
     for (let i = 0; i < this.maxConcurrency; i++) {
-      this.runningTasks.push(this.runTask());
+      const errorChannel = new TaskErrorChannel();
+      const taskRunner = new TaskRunner(i, errorChannel, () => this.runTask());
+
+      this.runningTasks.push(taskRunner.run());
+
+      errorChannel.onError((event) => {
+        const taskRunner = new TaskRunner(event.taskIndex, errorChannel, () => this.runTask());
+        this.runningTasks[event.taskIndex] = taskRunner.run();
+      });
     }
   }
 
   async stop(): Promise<void> {
     this.stopRequested = true;
 
-    const results = await Promise.allSettled(this.runningTasks);
+    await Promise.allSettled(this.runningTasks);
```

Comment:
we should probably just convert this back to a Promise.all for now. Believe i add this when we were talking but forgot to change it back